### PR TITLE
fix: HIVE job log contains 'java.lang.NullPointerException' error.(#4206)

### PR DESCRIPTION
**Pull request by CharlieYan24**
### What is the purpose of the change

fix: HIVE job log contains 'java.lang.NullPointerException' error.(#4206)

### Related issues/PRs

Related issues: #4206  #3906 

### Brief change log

- if driver.getPlan return null ,then invoke  driver.compile(realCode), to avoid java.lang.NullPointerException.
- make sure driver.compile(realCode) only execute once.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [x] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Comment by peacewong** [comment]
The account you submit needs to set up the GitHub username and email, GitHub may not count related contributions.
image

File: linkis-engineconn-plugins/hive/src/main/scala/org/apache/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala [hunk 1]
```diff
@@ -205,7 +205,15 @@ class HiveEngineConnExecutor(
       val startTime = System.currentTimeMillis()
       try {
         Utils.tryCatch {
-          val queryPlan = driver.getPlan
+          var queryPlan = driver.getPlan
```
Comment (peacewong) [inline linkis-engineconn-plugins/hive/src/main/scala/org/apache/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala:208]: This problem is caused by another fix. It is recommended to optimize and add parameters, whether to enable compilation.#3906
```
eg:
val ENABLE_HIVE_COMPILE =
    CommonVars[Boolean]("linkis.hive.enable.compile", true).getValue
```
image

**Comment by peacewong** [comment]
Can you add me as a WeChat friend? I pull you to the community development group, my account peaceAyton.

File: linkis-engineconn-plugins/hive/src/main/scala/org/apache/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala (hunk 1 above)
Comment (CharlieYan24) [reply linkis-engineconn-plugins/hive/src/main/scala/org/apache/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala:208]: > This problem is caused by another fix. It is recommended to optimize and add parameters, whether to enable compilation.#3906
> 
> ```
> eg:
> val ENABLE_HIVE_COMPILE =
>     CommonVars[Boolean]("linkis.hive.enable.compile", true).getValue
> ```
> 
> image

Thanks for your kind advice. I will optimize and add parameters, whether to enable compilation.

**Comment by CharlieYan24** [comment]
> Can you add me as a WeChat friend? I pull you to the community development group, my account peaceAyton.

Thanks a lot.

File: linkis-engineconn-plugins/hive/src/main/scala/org/apache/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala (hunk 1 above)
Comment (CharlieYan24) [reply linkis-engineconn-plugins/hive/src/main/scala/org/apache/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala]: > 

I have modify the code according to our talk, here is a brief explanation of code:
1, Compile the code.
2, Invoke driver.getPlan to get execution plan.
3, Invoke driver.run with parameter not compile ,avoiding recompile the query.

Comment (peacewong) [reply linkis-engineconn-plugins/hive/src/main/scala/org/apache/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala]: ok， Thanks a lot.

**Review by peacewong — APPROVED** [review]
LGTM.

**Comment by peacewong** [comment]
Thank you, the integration test has passed

**Pull request merged**